### PR TITLE
[mojo-stdlib] feature: Resize List smaller with single arg

### PR DESCRIPTION
**Pull request by mikowals**
Implements #2133

### CODE REVIEW

File: stdlib/src/collections/list.mojo [hunk 1]
```diff
@@ -265,12 +265,37 @@ struct List[T: CollectionElement](CollectionElement, Sized):
...
+                _ = (self.data + i).take_value()
+            for i in range(self.size, new_size):
+                (self.data + i).emplace_value(value)
+            self.size = new_size
+
+    @always_inline
+    fn resize(inout self, new_size: Int):
+        """Resizes the list to the given new size.
+
+        With no new value provided, the new size must be smaller than the current one. Elements at the end
```
Comment (JoeLoser) [inline stdlib/src/collections/list.mojo:282]: **Quibble** Here, we say "smaller than" but the `debug_assert` is "smaller than or equal".  I think we just want to fixup this doc string to say "smaller than or equal to".

**Review by JoeLoser — APPROVED** [review]
Great, thank you for your contribution!  Just one minor wording on the doc string comment.  Once that is fixed, I'm happy to merge this. 🎉

File: stdlib/src/collections/list.mojo (hunk 1 above)
Comment (mikowals) [reply stdlib/src/collections/list.mojo]: done.

Comment (JoeLoser) [reply stdlib/src/collections/list.mojo]: Thanks for the quick fix — merged!  Appreciate the contribution 🎉

**Pull request merged**